deployments/pulse3d/services/pulse3d/src/main.py

```diff
@@ -291,18 +317,27 @@ async def create_new_job(
 
         priority = 10
         async with request.state.pgpool.acquire() as con:
+
+            usage_quota = await check_customer_quota(con, customer_id, service)
+            if usage_quota["jobs_reached"]:
+                raise CustomerUsageQuotaReached()
```

Comment:
If this error is just going to get caught and then immediately return the response, we might as well just return the `UsageErrorResponse(usage_error=usage_quota)` here. I do think it would be useful though to include some logging so we can see in the svc logs if people are hitting routes and getting the `UsageErrorResponse`, so with that addition I think you can do it either way

Reply:
Ah good catch! I had initially done that when I was returning an error code instead of changing the response body with 200